### PR TITLE
AppChrome: Reduce re-renders for identical pageNavs

### PR DESCRIPTION
**Pull request by torkelo**
Many item pages pass new instances to Page component which always triggers a state
update by AppChrome service and triggers a re-render for AppChrome even though nothing
has really changed.

This adds some new logic to AppChromeService to ignore changes where
navModel or pageNav instance has changed but are really the same.

### CODE REVIEW

File: public/app/core/components/AppChrome/AppChrome.tsx [hunk 1]
```diff
@@ -20,6 +20,8 @@ export function AppChrome({ children }: Props) {
   const { chrome } = useGrafana();
   const state = chrome.useState();
 
+  console.log('AppChrome render');
```
Comment (ashharrison90) [inline public/app/core/components/AppChrome/AppChrome.tsx:23]: stray `console.log` here

**Review by ashharrison90 — COMMENTED** [review]
do we have a performance problem - does rerendering appchrome cause the whole page inside to rerender as well? 🤔 

if so i can see how this is necessary, even if it does feel a little brittle. 😬

File: public/app/core/components/AppChrome/AppChrome.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/core/components/AppChrome/AppChrome.tsx:23]: Fixed

**Comment by torkelo** [comment]
> do we have a performance problem - does rerendering appchrome cause the whole page inside to rerender as well?

Not noticeable but I really hate unnecessary react renders, and rendering AppChrome will trigger renders of
* CommandPalette
* SearchWrapper
* NavToolbar
* TopSearchBar 

These can be rendered 5 times when going into a DataSource edit page (due to navModel and pageNav always being a new instance)

**Review by ashharrison90 — APPROVED** [review]
think that's fair, just making sure we're not over optimising prematurely 👍

**Pull request merged**